File: code/python/runme.py
from swigmt import *;
import threading;

class DCallback(Callback):
    def __init__(self):
        Callback.__init__(self)

    def response(self):
        print("response from python in thread " + threading.current_thread().getName())

cb = DCallback()
caller = Caller(cb)
print("request from python in thread " + threading.current_thread().getName())
caller.request()

File: code/csharp/Runme.cs
using System;
using System.Threading;

namespace Swigmt
{
    public class DCallback : Callback
    {
        public override void response()
        {
            Console.WriteLine("response from c# in thread " + Thread.CurrentThread.ManagedThreadId);
        }
    }

    public class Runme
    {
        public static void Main()
        {
            using (Callback callback = new Callback())
            {
                Caller caller = new Caller(callback);
                Console.WriteLine("request from c# in thread " + Thread.CurrentThread.ManagedThreadId);
                caller.request();
            }
        }
    }
}
